src/main/resources/schemas/schemaFileSchema.json

```diff
@@ -118,6 +134,10 @@
                       {
                         "type": "string",
                         "pattern": "^extra"
+                      },
```

Comment:
Really we should merge the `userProperties` and `amplitudeProperties` into a single reference (unless there are differences I missed).

Reply:
amplitudeProperties lead to entries in the event_properties dictionary in the event payload while userProperties lead to entries in the user_properties dictionary. There are differences in how those two types of properties are presented in the Amplitude UI for querying.

Reply:
Right, what I'm referring to is the duplicated JSON schema code here. We can create a reference at the top of the schema file which can then be used for both of these properties, if they match.

Reply:
Will do, but in the process of doing this I noticed -- the definitions that are already there seem unused except `stringArray` and `nonNegativeInteger` (which is used in another, unused definition)